source/kraken/tests/apply_disruption_test.cpp

```diff
@@ -109,3 +117,855 @@ BOOST_FIXTURE_TEST_CASE(simple_train_cancellation, SimpleDataset) {
+            ("A1",              "23:10"_t, "23:11"_t)
+            ("stop_area:stop1", "24:10"_t, "24:11"_t)
+            ("stop_area:stop2", "25:20"_t, "25:21"_t)
+            ("A2",              "26:20"_t, "26:21"_t);
+
+    b.generate_dummy_basis();
+    b.finish();
+    b.data->pt_data->index();
+    b.data->build_raptor();
+    b.data->build_uri();
+    b.data->meta->production_date = bg::date_period(bg::date(2012,6,14), bg::days(7));
+
+    //we delete the stop_area 1, on vj passing midnight is impacted,
+
+    navitia::apply_disruption(b.impact(nt::RTLevel::Adapted, "stop_area_closed")
+                              .severity(nt::disruption::Effect::NO_SERVICE)
+                              .on(nt::Type_e::StopArea, "stop_area")
+                              .application_periods(btp("20120615T000000"_dt, "20120618T000000"_dt))
+                              .get_disruption(),
+                              *b.data->pt_data, *b.data->meta);
+
+    BOOST_REQUIRE_EQUAL(b.data->pt_data->lines.size(), 1);
+    BOOST_REQUIRE_EQUAL(b.data->pt_data->vehicle_journeys.size(), 3); // we have 1 vj that don't circulate at all :(
+
+    // Base VJ
+    auto vj  = b.data->pt_data->vehicle_journeys_map["vj:A:0"];
+    BOOST_CHECK_MESSAGE(ba::ends_with(vj->base_validity_pattern()->days.to_string(), "0111111"),
+            vj->base_validity_pattern()->days);
+    BOOST_CHECK_MESSAGE(ba::ends_with(vj->adapted_validity_pattern()->days.to_string(), "0111000"),
+            vj->adapted_validity_pattern()->days);
+
+    // Adapted VJ
+    vj  = b.data->pt_data->vehicle_journeys_map["vehicle_journey 0:Adapted:1:stop_area_closed"];
+    BOOST_CHECK_MESSAGE(ba::ends_with(vj->base_validity_pattern()->days.to_string(), "0000000"),
+            vj->base_validity_pattern()->days);
+    BOOST_CHECK_MESSAGE(ba::ends_with(vj->adapted_validity_pattern()->days.to_string(), "0000111"),
+            vj->adapted_validity_pattern()->days);
+
+    b.data->build_uri();
+    b.data->pt_data->index();
+    b.finish();
+    b.data->build_raptor();
+
+    navitia::routing::RAPTOR raptor(*(b.data));
+
+    auto compute = [&](nt::RTLevel level, const std::string& from, const std::string& to,
+            navitia::DateTime datetime, int day) {
+        return raptor.compute(b.data->pt_data->stop_areas_map.at(from), b.data->pt_data->stop_areas_map.at(to),
+                datetime, day, navitia::DateTimeUtils::inf, level, 2_min, true);
+    };
+    auto res = compute(nt::RTLevel::Adapted, "A1", "A2", "23:00"_t, 1);
+    BOOST_REQUIRE_EQUAL(res.size(), 1);
+    BOOST_REQUIRE_EQUAL(res[0].items[0].stop_times.size(), 2);
+    auto vp = res[0].items[0].stop_times[0]->vehicle_journey->adapted_validity_pattern();
+    BOOST_CHECK_MESSAGE(ba::ends_with(vp->days.to_string(), "0000111"),  vp->days.to_string());
+
+    navitia::delete_disruption("stop_area_closed", *b.data->pt_data, *b.data->meta);
+
+    check_vjs_without_disruptions(b.data->pt_data->vehicle_journeys);
```

Comment:
minor: I think it would be nice to add a journey computation after the removal too.